core/src/main/scala/kafka/log/Log.scala

```diff
@@ -304,6 +304,65 @@ class Log(@volatile var dir: File,
       s"log end offset $logEndOffset in ${time.milliseconds() - startMs} ms")
   }
 
+  // the high watermark offset value, in non-leader logs only its message offsets are kept
+  @volatile private[this] var _highWatermarkMetadata : LogOffsetMetadata = LogOffsetMetadata(0)
```

Comment:
Can we move this above with the rest of the fields? Also, I think we can get rid of the var `replicaHighWatermark`.

Reply:
Done.